internal/expr/prepare.go

```diff
@@ -1,43 +1,49 @@
 package expr
 
 import (
-	"bytes"
 	"fmt"
 	"reflect"
 	"sort"
-	"strconv"
 	"strings"
 )
 
-// PreparedExpr contains an SQL expression that is ready for execution.
+// PreparedExpr represents a valid SQLair statement.
 type PreparedExpr struct {
-	outputs []typeMember
-	inputs  []typeMember
-	sql     string
+	preparedParts []preparedPart
 }
 
-// SQL returns the SQL ready for execution.
-func (pe *PreparedExpr) SQL() string {
-	return pe.sql
+// preparedPart represents a part of a SQLair statement. It contains
+// information to generate the SQL for the part and to access Go types
+// referenced in the part.
+type preparedPart interface {
+	preparedPart()
 }
 
-const markerPrefix = "_sqlair_"
+// preparedOutput contains the columns to fetch from the database and
+// information about the Go values to read the results into.
+type preparedOutput struct {
+	columns []columnAccessor
```

Comment:
I would like to change this type to `string` instead of `columnAccessor` since at this point we have "prepared" the column and only need it for the SQL generation. I think this should be in a separate PR though.